python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -254,39 +278,157 @@ def _create_chat_message_content(
-        role: str | None = None,
+        stream_event: TextEvent | ContentBlockStopEvent | RawMessageDeltaEvent,
         metadata: dict[str, Any] = {},
     ) -> StreamingChatMessageContent:
-        """Create a streaming chat message content object from a choice."""
-        text_content = ""
-
-        if stream_event.delta and hasattr(stream_event.delta, "text"):
-            text_content = stream_event.delta.text
-
-        items: list[STREAMING_ITEM_TYPES] = [StreamingTextContent(choice_index=content_block_idx, text=text_content)]
-
+        """Create a streaming chat message content object from a content block."""
+        items: list[STREAMING_ITEM_TYPES] = []
         finish_reason = None
-        if isinstance(stream_event, RawMessageDeltaEvent):
-            if stream_event.delta.stop_reason:
-                finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[stream_event.delta.stop_reason]
 
-            metadata["usage"].completion_tokens = stream_event.usage.output_tokens
+        if isinstance(stream_event, TextEvent):
+            items.append(StreamingTextContent(choice_index=0, text=stream_event.text))
+        elif (
+            isinstance(stream_event, ContentBlockStopEvent)
+            and hasattr(stream_event, "content_block")
+            and stream_event.content_block.type == "tool_use"
+        ):
+            tool_use_block = stream_event.content_block
+            items.append(
+                FunctionCallContent(
+                    id=tool_use_block.id,
+                    index=stream_event.index,
+                    name=tool_use_block.name,
+                    arguments=json.dumps(tool_use_block.input) if tool_use_block.input else None,
+                )
+            )
+        elif isinstance(stream_event, RawMessageDeltaEvent):
+            finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[str(stream_event.delta.stop_reason)]
+            metadata["usage"]["output_tokens"] = stream_event.usage.output_tokens
 
         return StreamingChatMessageContent(
-            choice_index=content_block_idx,
+            choice_index=0,
             inner_content=stream_event,
             ai_model_id=self.ai_model_id,
             metadata=metadata,
-            role=AuthorRole(role) if role else AuthorRole.ASSISTANT,
+            role=AuthorRole.ASSISTANT,
             finish_reason=finish_reason,
             items=items,
         )
+
+    def update_settings_from_function_call_configuration_anthropic(
+        self,
+        function_choice_configuration: FunctionCallChoiceConfiguration,
+        settings: "PromptExecutionSettings",
+        type: "FunctionChoiceType",
+    ) -> None:
+        """Update the settings from a FunctionChoiceConfiguration."""
+        if (
+            function_choice_configuration.available_functions
+            and hasattr(settings, "tools")
+            and hasattr(settings, "tool_choice")
+            and type != FunctionChoiceType.NONE
```

Comment:
I think the `type != FunctionChoiceType.NONE` check is fine here, but it'd be my wish to make sure we never even get to this point with `NONE` defined (perhaps the model validator to the rescue?).